app/frontend/src/components/Select.tsx

```diff
@@ -20,19 +20,41 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export default function Select({
+export default function Select<
+  T extends { [key: string]: string } | { [key: number]: string }
+>({
   id,
   children,
   className,
+  data,
   label,
   variant = "outlined",
   options,
   ...otherProps
 }: SelectProps & {
   id: string;
-  options?: { label: string; value: string | number }[];
+  options?: Extract<keyof T, string | number>[];
+  value?: T extends undefined ? string | number : keyof T;
+  data?: T;
```

Comment:
The reason I need both `options` and `data` is because the labels object contains more items than I want to display in the dropdown. I don't see how that would be different when using Record (which seems a better option here indeed) than an index signature.

Reply:
Oh, I see. Can you think of a better name for `data`? I wouldn't know the difference between it and `options` without looking at a previous usage or the source.

Reply:
what about 'optionLabelMap' ?

Reply:
Sounds good